node/impl/full/eth.go

```diff
@@ -2135,3 +2137,57 @@ func (g gasRewardSorter) Swap(i, j int) {
 func (g gasRewardSorter) Less(i, j int) bool {
 	return g[i].premium.Int.Cmp(g[j].premium.Int) == -1
 }
+
+func (a *EthModule) EthGetBlockReceipts(ctx context.Context, blockParam ethtypes.EthBlockNumberOrHash) ([]*api.EthTxReceipt, error) {
+	ts, err := getTipsetByEthBlockNumberOrHash(ctx, a.Chain, blockParam)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get tipset: %w", err)
+	}
+
+	messages, err := a.Chain.MessagesForTipset(ctx, ts)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get messages for tipset: %w", err)
+	}
+
+	// Get the next tipset where the messages are actually executed
+	nextTs, err := a.Chain.GetTipsetByHeight(ctx, ts.Height()+1, ts, false)
```

Comment:
```suggestion
	nextTs, err := a.Chain.GetTipsetByHeight(ctx, ts.Height()+1, nil, false)
```

The tipset argument is intended to serve as a starting point, by providing `nil` you're saying "just find it for me starting from the head". Providing `ts` says "find me the _next_ tipset, by starting at this one and walking backward", which won't work.